clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppEngineLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,124 @@
+import com.netflix.spinnaker.clouddriver.appengine.model.AppEngineTrafficSplit
+import com.netflix.spinnaker.clouddriver.appengine.provider.view.AppEngineClusterProvider
+import com.netflix.spinnaker.clouddriver.appengine.provider.view.AppEngineLoadBalancerProvider
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+
+@Slf4j
+class UpsertAppEngineLoadBalancerAtomicOperation implements AtomicOperation<Map> {
+  private static final String BASE_PHASE = "UPSERT_LOAD_BALANCER"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  private final UpsertAppEngineLoadBalancerDescription description
+
+  @Autowired
+  AppEngineLoadBalancerProvider appEngineLoadBalancerProvider
+
+  @Autowired
+  AppEngineClusterProvider appEngineClusterProvider
+
+  UpsertAppEngineLoadBalancerAtomicOperation(UpsertAppEngineLoadBalancerDescription description) {
+    this.description = description
+  }
+
+  /**
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertLoadBalancer": { "loadBalancerName": "second", "credentials": "my-appengine-account", "migrateTraffic": false, "split": { "shardBy": "COOKIE" } } } ]' localhost:7002/appengine/ops
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertLoadBalancer": { "loadBalancerName": "default", "credentials": "my-appengine-account", "migrateTraffic": false, "split": { "shardBy": "IP", "allocations": { "app-stack-detail-v000": "0.5", "app-stack-detail-v001": "0.5" } } } } ]' localhost:7002/appengine/ops
+   */
+  @Override
+  Map operate(List priorOutputs) {
+    task.updateStatus BASE_PHASE, "Initializing upsert of load balancer $description.loadBalancerName " +
+      "in $description.credentials.region..."
+
+    def credentials = description.credentials
+    def appengine = credentials.appengine
+    def loadBalancerName = description.loadBalancerName
+    def updateSplit = description.split
+
+    // TODO(dpeach): do validation for traffic migration.
+    def migrateTraffic = description.migrateTraffic
+
+    if (updateSplit.allocations) {
+      validateServerGroupsCanBeEnabled(updateSplit.allocations, loadBalancerName)
+    }
+
+    def ancestorLoadBalancer = appEngineLoadBalancerProvider.getLoadBalancer(credentials.name, loadBalancerName)
+    def override = copyAndOverrideAncestorSplit(ancestorLoadBalancer.split, updateSplit)
+
+    def service = new Service(split: new TrafficSplit(allocations: override.allocations, shardBy: override.shardBy.toString()))
+
+    appengine.apps().services().patch(credentials.project, loadBalancerName, service)
+      .setUpdateMask("split")
+      .setMigrateTraffic(migrateTraffic)
+      .execute()
+
+    // TODO(dpeach): write operation poller and poll for operation completion here.
```

Comment:
I don't think you need an operation poller here, it's up to Orca to determine when this has finished upserting.